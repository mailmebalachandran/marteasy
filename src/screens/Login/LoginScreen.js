import React, {Component} from 'react';
import {View, Image, ActivityIndicator} from 'react-native';
import TextBox from '../../components/TextBox/TextBox';
import Label from '../../components/Label/Label';
import styles from './styles';
import StatusBarComponent from '../../components/StatusBar/StatusBarComponent';
import ButtonComponent from '../../components/Button/Button';
import LoginAPI from '../../api/Login/LoginAPI';
import Validation from '../../validation/Login/LoginValidation';
import Toast, {DURATION} from 'react-native-easy-toast';
import {LOGO} from '../../assets/index';
import * as ThemeColor from '../../themes/colors';
import * as Constants from './constants';

class LoginScreen extends Component {
  state = {
    UserName: '',
    Password: '',
    IsLoaded: false,
  };

  onSubmitHandler = async () => {
    this.setState({IsLoaded: true});
    const userDetails = {
      username: this.state.UserName,
      password: this.state.Password,
    };
    let validationResult = Validation.LoginValidation(userDetails);
    if (validationResult.isValidated) {
      let result = await LoginAPI.LoginValidation(userDetails);
      if (!result.isValidated) {
        this.refs.toast.show(result.message, DURATION.LENGTH_LONG);
      } else {
        this.props.navigation.navigate('HomeScreen');
      }
      this.setState({IsLoaded: false});
    } else {
      this.refs.toast.show(validationResult.message, DURATION.LENGTH_LONG);
      this.setState({IsLoaded: false});
    }
  };

  render() {
    return (
      <View style={styles.containerStyle}>
        {this.state.IsLoaded && (
          <ActivityIndicator size="large" color={ThemeColor.DarkColor} />
        )}
        <StatusBarComponent styleType={0} />
        <View style={styles.logoStyle}>
          <Image source={LOGO} />
        </View>
        <Label labelValue={Constants.LABEL_USERNAME} />
        <TextBox
          placeHolderValue={Constants.PLACEHOLDER_USERNAME}
          textValue={this.state.UserName}
          IsHavingIcon={true}
          iconName="user"
          iconSize={25}
          iconColor={ThemeColor.DarkColor}
          secureText={false}
          textStyle={{fontSize: 10}}
          autoCapitalize="none"
          onChangedTextHandler={text => {
            this.setState({UserName: text});
          }}
        />
        <Label labelValue={Constants.LABEL_PASSWORD} />
        <TextBox
          placeHolderValue={Constants.PLACEHOLDER_PASSWORD}
          textValue={this.state.Password}
          IsHavingIcon={true}
          iconName="lock"
          iconSize={25}
          iconColor={ThemeColor.DarkColor}
          secureText={true}
          autoCapitalize="none"
          onChangedTextHandler={text => {
            this.setState({Password: text});
          }}
        />
        <ButtonComponent
          titleValue={Constants.BUTTON_LOGIN}
          onPressHandler={this.onSubmitHandler}
        />
        <ButtonComponent
          titleValue={Constants.BUTTON_CANCEL}
          onPressHandler={() => {
            this.props.navigation.navigate('HomeScreen');
          }}
        />
        <Toast
          ref="toast"
          style={{backgroundColor: '#dfdfdf'}}
          position="top"
          positionValue={100}
          fadeInDuration={750}
          fadeOutDuration={1000}
          opacity={0.8}
          textStyle={{color: 'black'}}
        />
      </View>
    );
  }
}

export default LoginScreen;
