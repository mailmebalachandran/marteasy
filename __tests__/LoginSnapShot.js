import React from 'react';
import LoginScreen from '../src/screens/Login/LoginScreen';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});