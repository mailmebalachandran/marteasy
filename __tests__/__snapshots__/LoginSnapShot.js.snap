// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <View />
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginTop": "30%",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/images/logo.png",
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#087f23",
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
      >
        Username
      </Text>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#087f23",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            editable={true}
            onChangeText={[Function]}
            placeholder="Username"
            placeholderTextColor="#e0e0e0"
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 14,
                "minHeight": 40,
              }
            }
            testID="username"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#087f23",
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
      >
        Password
      </Text>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#087f23",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            editable={true}
            onChangeText={[Function]}
            placeholder="Password"
            placeholderTextColor="#e0e0e0"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 14,
                "minHeight": 40,
              }
            }
            testID="password"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "margin": 0.1,
          }
        }
      >
        <Text
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "textDecorationLine": "underline",
            }
          }
        >
          Forgot Password?
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 0.14,
            "flexDirection": "row",
            "marginTop": "5%",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 3,
                "margin": 10,
                "overflow": "hidden",
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#087f23",
                    "borderRadius": 3,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#087f23",
                      "fontSize": 18,
                      "paddingBottom": 1,
                      "paddingTop": 2,
                      "textAlign": "center",
                    }
                  }
                >
                  Login
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 3,
                "margin": 10,
                "overflow": "hidden",
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderColor": "#087f23",
                    "borderRadius": 3,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#087f23",
                      "fontSize": 18,
                      "paddingBottom": 1,
                      "paddingTop": 2,
                      "textAlign": "center",
                    }
                  }
                >
                  Cancel
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.01,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": "10%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
              "fontSize": 17,
            }
          }
        >
          New User
        </Text>
        <Text
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "color": "green",
              "fontSize": 18,
              "marginLeft": 5,
              "textDecorationLine": "underline",
            }
          }
        >
          SignUp
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
